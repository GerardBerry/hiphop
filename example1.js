"use strict"

var reactive = require("./reactive-kernel.js");

/*
  <ReactiveMachine>
     <pause />
     <emit signal=${s} />
     <present signal=${s}>
        <emit signal=${t} />
     </present>
  </ReactiveMachine>
*/

var sigS = new reactive.Signal("S", false, function() {
   console.out("EMIT S");
});

var sigT = new reactive.Signal("T", false, function() {
   console.out("EMIT T");
});

var emitS = new reactive.Emit(sigS);
var emitT = new reactive.Emit(sigT);
var present = new reactive.Present(sigS, emitT);
var pause = new Pause();
var seq = new Sequence(pause, emitS, present);
var machine = new ReactiveMachine(seq);

for (var i = 0; i < 5; i++)
   machime.react(i);
