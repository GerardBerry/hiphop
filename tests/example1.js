"use strict"

var reactive = require("../reactive-kernel.js");
var inspector = require("../inspector.js");

/*
  <ReactiveMachine>
     <pause />
     <emit signal=${s} />
     <present signal=${s}>
        <emit signal=${t} />
     </present>
  </ReactiveMachine>
*/

var sigS = new reactive.Signal("S", false, function() {
   console.log("EMIT S");
});

var sigT = new reactive.Signal("T", false, function() {
   console.log("EMIT T");
});

var emitS = new reactive.Emit(sigS);
var emitT = new reactive.Emit(sigT);
var present = new reactive.Present(sigS, emitT);
var pause = new reactive.Pause();
var seq = new reactive.Sequence(pause, emitS, present);
var machine = new reactive.ReactiveMachine(seq);

//(new inspector.Inspector(machine)).inspect();
for (var i = 0; i < 5; i++)
   machine.react(i);

