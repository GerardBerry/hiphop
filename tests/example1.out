example1> ;
--- Output: 
example1> ;
--- Output: T 
example1> ;
--- Output: 
example1> ;
--- Output: 
example1> ;
--- Output: 
example1> !reset;
--- Automaton example1 reset
example1> ;
--- Output: 
example1> ;
--- Output: T 
example1> ;
--- Output: 
example1> ;
--- Output: 
example1> ;
--- Output: 
